java/ingest/ingest-batcher-submitterv2/src/main/java/sleeper/ingest/batcher/submitterv2/IngestBatcherSubmitter.java

```diff
@@ -15,81 +15,119 @@
-                .map(file -> {
-                    String filePath = HadoopPathUtils.getRequestPath(file);
-                    LOGGER.info("Deserialised ingest request for file {} with size {} to table {}",
-                            filePath, NumberFormatUtils.formatBytes(file.getLen()), table);
-                    return IngestBatcherTrackedFile.builder()
-                            .file(filePath)
-                            .fileSizeBytes(file.getLen())
-                            .tableId(table.getTableUniqueId())
-                            .receivedTime(receivedTime)
-                            .build();
-                })
-                .collect(Collectors.toList());
+        String tableID = tableIndex.getTableByName(request.tableName())
+                .orElseThrow(() -> TableNotFoundException.withTableName(request.tableName())).getTableUniqueId();
+        List<IngestBatcherTrackedFile> list = new ArrayList<IngestBatcherTrackedFile>();
+
+        for (String filename : request.files()) {
+            if (!filename.contains("/")) {
+                list.addAll(getFilesByBucket(filename, tableID, receivedTime));
+            } else {
+                String[] filenameParts = filename.split("/");
+                String bucket = filenameParts[0];
+                String fileOrDirName = filenameParts[1];
+                if (fileOrDirName.contains(".")) {
+                    list.add(getIndividualFile(bucket, fileOrDirName, tableID, receivedTime));
+                } else {
+                    list.addAll(getFilesByDirectory(bucket, fileOrDirName, tableID, receivedTime));
+                }
+            }
+        }
+        return list;
+    }
+
+    private List<IngestBatcherTrackedFile> getFilesByBucket(String bucket, String tableID, Instant receivedTime) {
+        return getFilesByDirectory(bucket, "", tableID, receivedTime);
     }
 
+    private List<IngestBatcherTrackedFile> getFilesByDirectory(String bucket, String directory, String tableID, Instant receivedTime) {
+        List<IngestBatcherTrackedFile> list = new ArrayList<IngestBatcherTrackedFile>();
+        ListObjectsV2Iterable response = s3Client.listObjectsV2Paginator(
+                ListObjectsV2Request.builder()
+                        .bucket(bucket)
+                        .prefix(directory)
+                        .build());
+
+        for (ListObjectsV2Response page : response) {
+            page.contents().forEach((S3Object object) -> {
+                list.add(getIndividualFile(bucket, object.key(), tableID, receivedTime));
+            });
+        }
+        return list;
+    }
+
+    private IngestBatcherTrackedFile getIndividualFile(String bucket, String key, String tableID, Instant receivedTime) {
+        return buildTrackedFile(bucket + "/" + key, getFileSizeBites(bucket, key), tableID, receivedTime);
+    }
+
+    private Long getFileSizeBites(String bucket, String key) {
+        return s3Client.getObject(
```

Comment:
GetObject returns the contents of the file as well as the metadata. I think we need HeadObject to get just the metadata.